network/kademlia.go

```diff
@@ -569,51 +569,31 @@ func (k *Kademlia) callable(e *entry) bool {
 	return true
 }
 
-func (k *Kademlia) CloserPeerThanMeXOR(addr []byte) bool {
-	var ret bool
-
-	myDistance := make([]byte, 32)
-	chunk.XOR(myDistance, k.BaseAddr(), addr)
-
+// IsClosestTo returns true if self is the closest peer to addr among filtered peers
+// ie. return false iff there is a peer that
+// - filter(bzzpeer) == true AND
+// - pot.DistanceCmp(addr, peeraddress, selfaddress) == 1
+func (k *Kademlia) IsClosestTo(addr []byte, filter func(*BzzPeer) bool) (closest bool) {
+	myPo := chunk.Proximity(addr, k.BaseAddr())
```

Comment:
I think we should move the `Proximity` out of `Chunk` - it does not apply to a `Chunk` only, but also peers, messages...

Reply:
I don't care.